lib/search/index.ts

```diff
@@ -52,8 +55,14 @@ const tagsFromSearch = (query: string) => {
   return searchTags;
 };
 
+export let searchNotes: (
+  args: Partial<SearchState>,
+  maxResults: number
+) => [T.EntityId, T.Note | undefined][] = () => [];
```

Comment:
this `undefined` jumps off the screen - why would we return results for notes that don't exist?

should we not be filtering out those notes before returning them? what is a caller supposed to do if they get a note id without a note?

Reply:
@belcherj and I were of consensus that it would never return `undefined`, but TypeScript was unhappy, so we added this. Please do fix in a follow-up PR if you know the fix for it :)